programs/mpl-core/src/processor/burn.rs

```diff
@@ -138,9 +138,9 @@ pub(crate) fn burn_collection<'a>(
         ctx.accounts.collection,
         None,
         None,
-        CollectionV1::check_burn,
+        CollectionV1::check_burn_collection,
         PluginType::check_burn,
-        CollectionV1::validate_burn,
+        CollectionV1::validate_burn_collection,
         Plugin::validate_burn,
         Some(ExternalPluginAdapter::validate_burn),
         Some(HookableLifecycleEvent::Burn),
```

Comment:
Baseline: So `Plugin::validate_burn` and `ExternalPluginAdapter::validate_burn` will be called for plugins and external plugin adapters on the collection, and their state or control in general refers to both the collection and assets in the collection.  Here they are being specifically used in the `burn_collection` instruction, so they only refer to to the collection.

So one thing is if there's a frozen plugin on the collection it will block burning the collection, which is fine but probably irrelevant.

But what about external plugin adapter?  Do we want an Oracle to block burning a collection?  We could consider putting `None` in the external plugin adapter stuff here if we think those are only meant to refer to asset lifecycle events.

I think it would actually be the most academically proper here to have all the function pointers be `check_burn_collection`/`validate_burn_collection`, since that is the specific lifecycle event being validated.

It's probably not super relevant, since you can only burn empty collections at which time you most likely  don't care about plugins and external plugins any more anyways.  But wanted to just decide if we should make it the most academically proper, just for clarity in the code.

Reply:
Just thinking through this a little more and I have a counter-thought for what's academically proper. Our current core assertions are:
- Collections do not have lifecycle events
- No delegates, including the UpdateDelegate, can burn a collection
So playing devil's advocate, maybe the fact that we're doing validations in the burn_collection handler is actually incorrect and this should instead be a one-off instruction that asserts the authority is correct and the collection is empty, and no other validations. This would maintain our first assertion and would not counter the second unless we eventually need to reject or authorize burns in the future.

Reply:
>Collections do not have lifecycle events
I think if we add `CollectionV1::check_burn_collection` and we add `CollectionV1::validate_burn_collection`, then we are crossing a line saying collections actually do have lifecycle events.

Deciding not to cross the line seems ok with me, but seems like there's some edge cases to think about.  For example, you mention no delegates can burn a collection.  But what about if the collection is frozen?

Reply:
Resolved offline.  Decided collections will not have lifecycle events at this time.